channels/audin/client/audin_main.c

```diff
@@ -425,13 +425,16 @@ static BOOL audin_open_device(AUDIN_PLUGIN* audin, AUDIN_CHANNEL_CALLBACK* callb
 		format.cbSize = 0;
 		for (x = 0; x < ARRAYSIZE(samplerates); x++)
 		{
+			size_t y;
 			format.nSamplesPerSec = samplerates[x];
-			format.nChannels = audin->format->nChannels;
-			test = IFCALLRESULT(FALSE, audin->device->FormatSupported, audin->device, &format);
-			if (test)
-				break;
-			format.nChannels = 3 - format.nChannels;
-			test = IFCALLRESULT(FALSE, audin->device->FormatSupported, audin->device, &format);
+			for (y = audin->format->nChannels; y > 0; y--)
+			{
+				format.nChannels = audin->format->nChannels;
```

Comment:
If I understand what you wanted to do, you should write 
`format.nChannels = y` instead of `format.nChannels = audin->format->nChannels`
...otherwise, I don't see the point of the iterator y.

Anyway, this won't work well, because if `audin->format->nChannels` is 1, you won't test with 2 channels, and this was the purpose of the code you replace.

Reply:
ok, never commit before coffee, thank you @Pollux42 :)

Reply:
No problem, but have you got your coffee now?
Because I still don't see the point of this diff. :-)